components/canvas/webgl_mode/inprocess.rs

```diff
@@ -73,18 +77,29 @@ impl WebGLThreads {
 struct WebGLExternalImages {
     webrender_gl: Rc<dyn gl::Gl>,
     webgl_channel: WebGLSender<WebGLMsg>,
+    // Mapping between an IOSurface and the texture it is bound on the WR thread
+    textures: FnvHashMap<IOSurfaceId, gl::GLuint>,
     // Used to avoid creating a new channel on each received WebRender request.
     lock_channel: (
-        WebGLSender<(u32, Size2D<i32>, usize)>,
-        WebGLReceiver<(u32, Size2D<i32>, usize)>,
+        WebGLSender<WebGLLockMessage>,
+        WebGLReceiver<WebGLLockMessage>,
     ),
 }
 
+impl Drop for WebGLExternalImages {
+    fn drop(&mut self) {
+        for (_, texture_id) in &self.textures {
+            self.webrender_gl.delete_textures(&[*texture_id]);
```

Comment:
We will want to have a followup issue filed for deleting these textures when the corresponding webgl context is removed.